core/src/epicli/data/common/ansible/playbooks/roles/applications/tasks/deploy-k8s-app.yml

```diff
@@ -1,24 +1,47 @@
 ---
-- name: "{{ data.name }}: Set directory for deployment files"
-  set_fact:
-    deployment_tmp_dir: /tmp/epiphany/apps/{{ app_dir_name }}
+- name: Assert input parameters
+  assert:
+    that:
+      - deployment_tmp_root_dir is defined
+      - app_dir_name is defined
 
-- name: "{{ data.name }}: Create directory for deployment files"
-  file:
-    path: "{{ deployment_tmp_dir }}"
-    state: directory
-    owner: "{{ admin_user.name }}"
-    group: "{{ admin_user.name }}"
-
-- name: "{{ data.name }}: Deploy K8s files"
-  include_tasks: deploy-k8s-file.yml
+- name: Deploy application
   vars:
-    file_name: "{{ deployment_file }}"
-  loop: "{{ deployment_files }}"
-  loop_control:
-    loop_var: deployment_file
+    deployment_tmp_dir: "{{ deployment_tmp_root_dir }}/{{ app_dir_name }}"
+  always:
```

Comment:
This will make debugging more difficult. I would like to have files in place in order to run a failed command manually. Since files may contain secrets, maybe we could use `ansible-vault` for storing them and remove only when deployment succeeded.

Reply:
Please check in latest commits. :+1: